src/components/patient/PatientDetails.tsx

```diff
@@ -12,13 +13,27 @@ import ApiErrorMessage from '../error/ApiErrorMessage'
 import NotFound from '../error/NotFound'
 import PageHeader from '../shell/PageHeader'
 import AddressPanel from './AddressPanel'
+import ConfirmationModal from './ConfirmationModal'
 import DemographicsPanel from './DemographicsPanel'
 import IdentifiersPanel from './IdentifiersPanel'
 import RegisteringFacilityPanel from './RegisteringFacilityPanel'
 import RelationshipPanel from './RelationshipPanel'
 import SubHeading from './SubHeading'
 
+export interface UpdatedFields {
+  id: number
+  field: string
+  original: any
+  new: any
+}
+
 const PatientDetails = () => {
+  const [patientRecord, setPatientRecord] = useState<PatientRecord | undefined>(
+    undefined
+  )
+  const [isEditable, setIsEditable] = useState(false)
+  const [isModalVisible, setIsModalVisible] = useState(false)
+  const [updatedFields, setUpdatedFields] = useState<UpdatedFields[]>([])
```

Comment:
Let's have this rather an object where the keys are the field names.
```
type UpdateFields = {
  [fieldName: string]: { oldValue: , newValue }
}
```
This will allow you to easily update the state (instead of tables)